dts/bindings/pinctrl/nxp,kinetis-pinctrl.yaml

```diff
@@ -0,0 +1,78 @@
+# Copyright (c) 2022, NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Kinetis pinctrl node. This node will define pin configurations in pin groups,
+  and has the 'pinctrl' node identifier in the SOC's devicetree. Each group
+  within the pin configuration defines the pin configuration for a peripheral,
+  and each numbered subgroup in the pin group defines all the pins for that
+  peripheral with the same configuration properties. The 'pins' property in
+  a group selects the pins to be configured, and the remaining properties set
+  configuration values for those pins. Here is an example group for UART0 pins:
+
+  uart0_default: uart0_default {
+    group0 {
+      pins = <&UART0_RX_PTB16
+        &UART0_TX_PTB17>;
+      drive-strength = "low";
+      slew-rate = "fast";
+      };
+  };
+
+  If only the required properties are supplied, the pin configuration register
+  will be assigned the following values:
+  PCR_PS=0,
+  PCR_PE=0,
+  PCR_ODE=0,
+  PCR_SRE=<slew-rate selection>,
+  PCR_DSE=<drive-strength selection>,
+  PCR_PFE=0
+
+compatible: "nxp,kinetis-pinctrl"
+
+include:
+  - name: base.yaml
+  - name: pincfg-node-group.yaml
+    child-binding:
+      child-binding:
+        property-allowlist:
+          - drive-open-drain
+          - bias-pull-up
+          - bias-pull-down
+
+child-binding:
+  description: Kinetis pin controller pin group
+  child-binding:
+    description: |
+      Kinetis pin controller pin configuration node
+    properties:
+      pins:
```

Comment:
nit: should likely be named `pinmux`, since this field describes the muxing right?